crates/view/src/worker.rs

```diff
@@ -217,10 +217,17 @@ impl Worker {
             }
         });
 
+        let mut expected_height = start_height;
+
         while let Some(block) = buffered_stream.recv().await {
             let block: CompactBlock = block?.try_into()?;
 
             let height = block.height;
+            if height < expected_height {
```

Comment:
Why not enforce consecutive ordering while we're at it? That "synergize" well with hardcoding truncated CB hashes later